src/goto-symex/goto_symex.cpp

```diff
@@ -665,3 +669,57 @@ bool goto_symext::constant_propagate_delete_char_at(
 
   return true;
 }
+
+bool goto_symext::constant_propagate_trim(
+  statet &state,
+  symex_assignt &symex_assign,
+  const function_application_exprt &f_l1)
+{
+  const auto &f_type = to_mathematical_function_type(f_l1.function().type());
+  const auto &length_type = f_type.domain().at(0);
+  const auto &char_type = to_pointer_type(f_type.domain().at(1)).subtype();
+
+  const refined_string_exprt &s = to_string_expr(f_l1.arguments().at(2));
+  const auto s_data_opt = try_evaluate_constant_string(state, s.content());
+
+  if(!s_data_opt)
+    return false;
+
+  const array_exprt &string_data = s_data_opt->get();
+  auto &operands = string_data.operands();
+
+  auto is_not_whitespace = [](const exprt &expr) {
+    auto &constant_value = to_constant_expr(expr);
+    char character = numeric_cast_v<int>(constant_value);
```

Comment:
numeric_cast_v<char> if possible otherwise use narrow, but actually the next operation doesn't require a char and could directly be done on an mp_integer